packages/flutter_tools/test/commands.shard/hermetic/build_ipa_test.dart

```diff
@@ -735,6 +748,99 @@ void main() {
     Platform: () => macosPlatform,
     XcodeProjectInterpreter: () => FakeXcodeProjectInterpreterWithBuildSettings(),
   });
+
+  testUsingContext(
+      'Validate basic Xcode settings with missing settings', () async {
+
+    const String plistPath = 'build/ios/archive/Runner.xcarchive/Products/Applications/Runner.app/Info.plist';
+    fakeProcessManager.addCommands(<FakeCommand>[
+      xattrCommand,
+      setUpFakeXcodeBuildHandler(onRun: () {
+        fileSystem.file(plistPath).createSync(recursive: true);
+      }),
+      exportArchiveCommand(exportOptionsPlist: _exportOptionsPlist),
+    ]);
+
+    createMinimalMockProjectFiles();
+
+    plistUtils.fileContents[plistPath] = <String,String>{
+      'CFBundleIdentifier': 'io.flutter.someProject',
+    };
+
+    final BuildCommand command = BuildCommand();
+    await createTestCommandRunner(command).run(
+        <String>['build', 'ipa', '--no-pub']);
+
+    expect(
+        testLogger.statusText,
+        contains(
+            '┌─ App Settings ──────────────────────────────────────────────────────────────────────┐\n'
+                '│ Version: Missing                                                                    │\n'
+                '│ Build: Missing                                                                      │\n'
+                '│ Display Name: Missing                                                               │\n'
+                '│ Deployment Target: Missing                                                          │\n'
+                '│ Bundle Identifier: io.flutter.someProject                                           │\n'
+                '│                                                                                     │\n'
+                '│ You must set up the missing settings                                                │\n'
+                '│ Instructions: https://docs.flutter.dev/deployment/ios#review-xcode-project-settings │\n'
+                '└─────────────────────────────────────────────────────────────────────────────────────┘'
+        )
+    );
+  }, overrides: <Type, Generator>{
+    FileSystem: () => fileSystem,
+    ProcessManager: () => fakeProcessManager,
+    Platform: () => macosPlatform,
+    XcodeProjectInterpreter: () =>
+        FakeXcodeProjectInterpreterWithBuildSettings(),
```

Comment:
@christopherfujino I hate to ask you religious questions but what's the recommended line length for the tool package?  It's not 80, right?

Reply:
ye ask and ye shall receiveth, https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#prefer-a-maximum-line-length-of-80-characters:

> Aim for a maximum line length of roughly 80 characters, but prefer going over if breaking the line would make it less readable, or if it would make the line less consistent with other nearby lines. Prefer avoiding line breaks after assignment operators.

Of course the devil lies in "less readable"...